service/Abstractions/Models/MemoryFilter.cs

```diff
@@ -17,18 +24,75 @@ public MemoryFilter ByTag(string name, string value)
         return this;
     }
 
+    public MemoryFilter ByNotTag(string name, string value)
+    {
+        this._notTags.Add(name, value);
+        return this;
+    }
+
     public MemoryFilter ByDocument(string docId)
     {
         this.Add(Constants.ReservedDocumentIdTag, docId);
         return this;
     }
 
+    /// <summary>
+    /// Get all the filters you need to put into the query
+    /// </summary>
+    /// <returns></returns>
     public IEnumerable<KeyValuePair<string, string?>> GetFilters()
     {
         return this.ToKeyValueList();
     }
+
+    /// <summary>
+    /// Gets all the filters that needs to be put as not into  the query
+    /// </summary>
+    /// <returns></returns>
+    public IEnumerable<KeyValuePair<string, string?>> GetNotFilters()
```

Comment:
this is never used, should we drop it?

Reply:
Correct, it is a leftover from an implementation I did in my internal project, then the code gets refactored so that method is not really used anymore.